### PR TITLE
Add init container to ensure Vault token

### PR DESCRIPTION
**Pull request by kubasobon**
Towards https://github.com/giantswarm/giantswarm/issues/14611

## Checklist

- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: helm/config-controller/templates/deployment.yaml [hunk 1]
```diff
@@ -39,6 +39,15 @@ spec:
       securityContext:
         runAsUser: {{ .Values.pod.user.id }}
         runAsGroup: {{ .Values.pod.group.id }}
+      initContainers:
+      - args:
+        - --vault-address={{ .Values.Installation.V1.Auth.Vault.Address }}
+        - --vault-role=cert-operator
```
Comment (corest) [inline helm/config-controller/templates/deployment.yaml:45]:
```suggestion
        - --vault-role=config-controller
```

**Review by corest — APPROVED** [review]
lgtm
You also need to mount secret in app to get access to it. E.g. https://github.com/giantswarm/cert-operator/blob/master/helm/cert-operator/templates/deployment.yaml#L69-L74

And also whole this thing will work only after vault configuration

**Comment by kubasobon** [comment]
> You also need to mount secret in app to get access to it. E.g. https://github.com/giantswarm/cert-operator/blob/master/helm/cert-operator/templates/deployment.yaml#L69-L74
I'm adding this part in #34 already.

> And also whole this thing will work only after vault configuration
What needs to be configured, specifically? The `auth/kubernetes/role/config-controller ` role mentioned [here](https://intranet.giantswarm.io/docs/support-and-ops/installation-guide-for-giantnetes/vault-kubernetes-auth-backend/)?

**Comment by kubasobon** [comment]
Forgot to mention: @corest :point_up:

**Pull request merged**